File: 1288-remove-covered-intervals/1288-remove-covered-intervals.py
class Solution:
    def removeCoveredIntervals(self, intervals: List[List[int]]) -> int:
        intervals.sort(key=lambda x: x[0])
        res, prev = 1, intervals[0]
        for i in range(1, len(intervals)):
            start, end = intervals[i]
            if start > prev[0] and end > prev[1]:
                res += 1
                prev[0] = start
            prev[1] = max(prev[1], end)
        return res

File: 1288-remove-covered-intervals/1288-remove-covered-intervals.js
/**
 * @param {number[][]} intervals
 * @return {number}
 */
var removeCoveredIntervals = function(intervals) {
    intervals.sort(([a], [b]) => a - b)
    let res = 0, start = -1, end = -1
    for (const [s, e] of intervals) {
        if (s > start && e > end) {
            res++
            start = s
        }
        end = Math.max(end, e)
    }
    return res
};